mesonbuild/wrap/wrap.py

```diff
@@ -228,6 +242,40 @@ def get_hg(self, p):
                 subprocess.check_call(['hg', 'checkout', revno],
                                       cwd=checkoutdir)
 
+    def get_build_file(self, p):
+        build_file = p.get('build_file')
+        print('Looking for %s' % build_file)
+        checkoutdir = os.path.join(self.subdir_root, p.get('directory'))
+        bf_name = os.path.basename(build_file)
+        dst_file = os.path.join(checkoutdir,"meson.build")
+        url = urlparse(build_file)
+        buff = []
+
+        # Get data buffer depending on location prefix (http(s))
+        if url.netloc:
+            buff = self.get_data(build_file)
+        else:
+            src_file = os.path.join(self.subdir_root, build_file)
+            if not os.path.exists(src_file):
+                print('Could not find {}'.format(src_file))
+                raise RuntimeError('Build file does not exist! %s' % src_file)
+            try:
+                with open(src_file) as s:
+                    buff = s.read()
+            except Exception as e:
+                print('Could not read {}'.format(src_file))
+                raise RuntimeError('Could not copy build file from %s\n%s'
+                % (src_file, e))
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->